include/alpaka/queue/cuda_hip/QueueUniformCudaHipRtBase.hpp

```diff
@@ -25,111 +27,132 @@
+            }
+            QueueUniformCudaHipRtImpl(QueueUniformCudaHipRtImpl&&) = default;
+            auto operator=(QueueUniformCudaHipRtImpl&&) -> QueueUniformCudaHipRtImpl& = delete;
+            ALPAKA_FN_HOST ~QueueUniformCudaHipRtImpl()
+            {
+                ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
 
-                    // In case the device is still doing work in the queue when cuda/hip-StreamDestroy() is called, the
-                    // function will return immediately and the resources associated with queue will be released
-                    // automatically once the device has completed all work in queue.
-                    // -> No need to synchronize here.
-                    ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_NOEXCEPT(ALPAKA_API_PREFIX(StreamDestroy)(m_UniformCudaHipQueue));
-                }
+                // In case the device is still doing work in the queue when cuda/hip-StreamDestroy() is called, the
+                // function will return immediately and the resources associated with queue will be released
+                // automatically once the device has completed all work in queue.
+                // -> No need to synchronize here.
+                ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_NOEXCEPT(ALPAKA_API_PREFIX(StreamDestroy)(m_UniformCudaHipQueue));
+            }
 
-                [[nodiscard]] auto getNativeHandle() const noexcept
-                {
-                    return m_UniformCudaHipQueue;
-                }
+            [[nodiscard]] auto getNativeHandle() const noexcept
+            {
+                return m_UniformCudaHipQueue;
+            }
 
-            public:
-                DevUniformCudaHipRt const m_dev; //!< The device this queue is bound to.
-            private:
-                ALPAKA_API_PREFIX(Stream_t) m_UniformCudaHipQueue;
-            };
+        public:
+            DevUniformCudaHipRt const m_dev; //!< The device this queue is bound to.
+        private:
+            ALPAKA_API_PREFIX(Stream_t) m_UniformCudaHipQueue;
+        };
 
-            //! The CUDA RT blocking queue.
-            class QueueUniformCudaHipRtBase
-                : public concepts::Implements<ConceptCurrentThreadWaitFor, QueueUniformCudaHipRtBase>
-                , public concepts::Implements<ConceptQueue, QueueUniformCudaHipRtBase>
-                , public concepts::Implements<ConceptGetDev, QueueUniformCudaHipRtBase>
+        //! The CUDA/HIP RT queue.
+        template<bool TBlocking>
+        class QueueUniformCudaHipRtBase final
+            : public concepts::Implements<ConceptCurrentThreadWaitFor, QueueUniformCudaHipRtBase<TBlocking>>
+            , public concepts::Implements<ConceptQueue, QueueUniformCudaHipRtBase<TBlocking>>
+            , public concepts::Implements<ConceptGetDev, QueueUniformCudaHipRtBase<TBlocking>>
+        {
+        public:
+            ALPAKA_FN_HOST QueueUniformCudaHipRtBase<TBlocking>(DevUniformCudaHipRt const& dev)
```

Comment:
```suggestion
            ALPAKA_FN_HOST QueueUniformCudaHipRtBase(DevUniformCudaHipRt const& dev)
```
Providing template arguments should not be necessary on constructors. Same for the equality operators.

Reply:
Right, will fix.